client/client.go

```diff
@@ -579,6 +803,50 @@ func (c *Client) Call(ctx context.Context, procedure string, options wamp.Dict,
 
 	switch msg := msg.(type) {
 	case *wamp.Result:
+
+		// Check and handle Payload PassThru Mode
+		// @see https://wamp-proto.org/wamp_latest_ietf.html#name-payload-passthru-mode
+		if pptScheme, _ := msg.Details[wamp.OptPPTScheme].(string); pptScheme != "" {
+			// Let's check: was ppt feature announced by dealer?
+			if !c.sess.HasFeature(wamp.RoleDealer, wamp.FeaturePayloadPassthruMode) {
+				// It's protocol violation, so we need to abort connection
+				abortMsg := wamp.Abort{Reason: wamp.ErrProtocolViolation}
+				abortMsg.Details = wamp.Dict{
+					"error": ErrPPTNotSupported.Error(),
+				}
+				abortMsg.Details[wamp.OptMessage] = "Peer is trying to use Payload Passthru Mode while it was not announced during HELLO handshake"
+				c.sess.Peer.Send(&abortMsg)
+				c.sess.Peer.Close()
+				return nil, ErrPPTNotSupported
+			}
+
+			switch {
+			case pptScheme == "wamp" || pptScheme == "mqtt" || strings.HasPrefix(pptScheme, "x_"):
+				// passthru of wamp and mqtt is supported by default, optionally a custom ppt scheme
+				// is also allowed
+			default:
+				return nil, ErrPPTSchemeInvalid
+			}
```

Comment:
used in a couple of places, could be:
```
func isPPTSchemeValid(pptScheme string) bool {
	// passthru of wamp and mqtt is supported by default, optionally a custom ppt scheme
	// is also allowed
	return pptScheme == "wamp" || pptScheme == "mqtt" || strings.HasPrefix(pptScheme, "x_")
}
```

and then the call sites are nice and self explanatory:
```
if !isPPTSchemeValid(pptScheme) {
	return nil, ErrPPTSchemeInvalid
}
```

Reply:
yeah. Agree! Thnx!